pytest_httpserver/httpserver.py

```diff
@@ -693,6 +677,219 @@ def create_matcher(self, *args, **kwargs) -> RequestMatcher:
+            if isinstance(assertion, AssertionError):
+                raise assertion
+
+            raise AssertionError(assertion)
+
+    def check_handler_errors(self):
+        """
+        Re-Raises any errors caused in request handlers
+
+        The first error raised by a handler will be re-raised here, and then
+        removed from the list.
+        """
+        if self.handler_errors:
+            raise self.handler_errors.pop(0)
+
+    def respond_nohandler(self, request: Request, extra_message: str = ""):
+        """
+        Add a 'no handler' assertion.
+
+        This method is called when the server wasn't able to find any handler to serve the request.
+        As the result, there's an assertion added (which can be raised by :py:meth:`check_assertions`).
+
+        """
+        text = "No handler found for request {!r}.\n".format(request)
+        self.add_assertion(text + extra_message)
+        return Response("No handler found for this request", self.no_handler_status_code)
+
+    @abc.abstractmethod
+    def dispatch(self, request: Request) -> Response:
+        """
+        Dispatch a request to the appropriate request handler.
+
+        :param request: the request object from the werkzeug library
+        :return: the response object what the handler responded, or a response which contains the error
+        """
+        pass
+
+    @Request.application  # type: ignore
+    def application(self, request: Request):
+        """
+        Entry point of werkzeug.
+
+        This method is called for each request, and it then calls the undecorated
+        :py:meth:`dispatch` method to serve the request.
+
+        :param request: the request object from the werkzeug library
+        :return: the response object what the dispatch returned
+        """
+        request.get_data()
+        response = self.dispatch(request)
+        self.log.append((request, response))
+        return response
+
+    def __enter__(self):
+        """
+        Provide the context API
+
+        It starts the server in a thread if the server is not already running.
+        """
+
+        if not self.is_running():
+            self.start()
+        return self
+
+    def __exit__(self, *args, **kwargs):
+        """
+        Provide the context API
+
+        It stops the server if the server is running.
+        Please note that depending on the internal things of werkzeug, it may take 0.5 seconds.
+        """
+        if self.is_running():
+            self.stop()
+
+
+class HTTPServer(HttpServerBase):  # pylint: disable=too-many-instance-attributes
+    """
+    Server instance which manages handlers to serve pre-defined requests.
+
+    :param default_waiting_settings: the waiting settings object to use as default settings for :py:meth:`wait` context
+        manager
+    """
+
+    def __init__(
+        self,
+        default_waiting_settings: Optional[WaitingSettings] = None,
+        **kwargs,
```

Comment:
Here, also, please retain the original documentation and replace `**kwargs` with parameters.
Strictly speaking, it also breaks the API in the case when someone specified arguments by position.

Reply:
ok.